src/devhostagent.restorationjob/RestorationJobApp.cs

```diff
@@ -142,10 +144,10 @@ private async Task<ExitCode> ExecuteInnerAsync(CancellationToken cancellationTok
                             perfLogger.SetProperty(HasConnectedClients, false);
 
                             var disconnectedTimeSpan = DateTimeOffset.Now - lastPingWithSessions;
-                            if (disconnectedTimeSpan != null && disconnectedTimeSpan.Value > _restorationJobEnvironmentVariables.RestoreTimeout)
+                            if ((disconnectedTimeSpan != null && disconnectedTimeSpan.Value > _restorationJobEnvironmentVariables.RestoreTimeout) || (lastPingWithSessions == null && count > 2))
                             {
                                 // Restore workload
-                                _log.Info($"Agent has no connected sessions for {disconnectedTimeSpan.Value:g}. Restoring...");

```

Comment:
removal or :g intentional?

Reply:
Yes this was causing not valid time error, but after refactoring the code, the purpose of changing this statement is not needed.